lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js

```diff
@@ -0,0 +1,84 @@
+const MapSegmentEditCapability = require("../../../core/capabilities/MapSegmentEditCapability");
+const ViomiMapParser = require("../ViomiMapParser");
+
+class ViomiMapSegmentEditCapability extends MapSegmentEditCapability {
+    /**
+     *
+     * @param {object} options
+     * @param {import("../../../core/ValetudoRobot")|any} options.robot
+     *
+     * @param {string} options.lang the default language for the generated room names
+     *
+     * @param {object} options.mapActions
+     * @param {number} options.mapActions.JOIN_SEGMENT_TYPE
+     * @param {number} options.mapActions.SPLIT_SEGMENT_TYPE
+     *
+     */
+    constructor(options) {
+        super(options);
+
+        this.lang = options.lang;
+        this.mapActions = options.mapActions;
+    }
+
+    /**
+     * Rounds and formats a point for use in viomi params.
+     *
+     * @param {object} point
+     * @param {number} point.x
+     * @param {number} point.y
+     */
+    pointToViomiString(point) {
+        const roundedX = Math.round((point.x + Number.EPSILON) * 100) / 100;
+        const roundedY = Math.round((point.y + Number.EPSILON) * 100) / 100;
+        return `${roundedX}_${roundedY}`;
+    }
+
+    /**
+     * @param {import("../../../entities/core/ValetudoMapSegment")} segmentA
+     * @param {import("../../../entities/core/ValetudoMapSegment")} segmentB
+     * @returns {Promise<void>}
+     */
+    async joinSegments(segmentA, segmentB) {
+        await this.robot.sendCommand("arrange_room", {
```

Comment:
If the currentMapId is required for this to work, it should throw if it's not available.

Another approach could be to poll the status before. Not sure what would be the best approach here though